core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,828 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for exploration models"""
+
+from __future__ import annotations
+
+import math
+
+from core.domain import exp_fetchers
+from core.domain import html_cleaner
+from core.domain import state_domain
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import exp_models
+
+datastore_services = models.Registry.import_datastore_services()
+
+(exp_models,) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class ExpStateValidationJob(base_jobs.JobBase):
+    """Job that tests the general state, rte and interaction validation"""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        total_explorations = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+        )
+
+        combine_exp_ids_and_states = (
+            total_explorations
+            | 'Combine exp id and states' >> beam.Map(
+                lambda exp: (exp.id, exp.states, exp.created_on))
+        )
+
+        invalid_exps_with_errored_state_rte_values = (
+            combine_exp_ids_and_states
+            | 'Get invalid state rte values' >> beam.Map(
+                lambda objects: (
+                    objects[0],
+                    self.filter_invalid_state_rte_values(objects[1]),
+                    objects[2]))
+            | 'Remove empty values for rte' >> beam.Map(
+                lambda objects: (
+                    objects[0], self.remove_empty_values(objects[1]),
+                    objects[2]))
+            | 'Split date and time' >> beam.Map(
+                lambda objects: (
+                    objects[0], objects[1], objects[2].date()
+                )
+            )
+        )
```

Comment:
This is a bit confusing, creating tuple and then referencing it just by indexes is hard to track, can you use some different approach, or use `MapTuple`

Reply:
ditto below

Reply:
Done